### PR TITLE
Support multiple drilldown searches response 

### PR DESCRIPTION
**Pull request by ShacharKidor**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [CIAC-9888](https://jira-dc.paloaltonetworks.com/browse/CIAC-9888).

## Description
Added support for multiple drill-down searches enrichment in SplunkPy integration. 
This feature was added to Splunk in Enterprise Security v7.2.0.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by ShacharKidor** [comment]
Hi @yuvalbenshalom, 
Please review the PR. 
About the delicate change we discussed (in the `submitted()` function - line 707), I investigated that again and now have a better understanding. I changed the condition there and added a similar change to the `failed_to_submit()` function (line 725). 
Let's talk about it again once you get there.  

Please let me know if something is unclear or needs to be improved. 
Thanks!

**Comment by ShirleyDenkberg** [comment]
@ilappe @yuvalbenshalom Doc review completed.

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 1]
```diff
@@ -656,22 +669,67 @@ def to_incident(self, mapper: UserMappingObject, comment_tag_to_splunk: str, com
...
+        # 1. First condition - if any of the notable's enrichments is 'in progress', it means that it was submitted to splunk.
+        # 2. Second condition - The ENABLED_ENRICHMENTS list contains the enrichment types that the user wants to enrich.
+        # According to the logic of the submit_notable() function, in a normal situation (where the code wasn't interrupted)
+        # the notable.enrichments list should include an enrichment object for each enrichment type that exist in the
+        # ENABLED_ENRICHMENTS list. That is because in the submit_notable() function we always add Enrichments objects to the
+        # notable.enrichments list regardless their statuses (failed\success). So if the function had finished it's run without
+        # any interruption we will have at least one enrichment object for each enrichment type (for drilldown enrichment we could
+        # have more than one enrichment object - in a case of multiple drilldown searches enrichment).
+
+        # TODO: consult with Yuval regarding the delicate change in the condition here.
```
Comment (yuvalbenshalom) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:726]:
```suggestion
```

File: Packs/SplunkPy/Scripts/SplunkShowDrilldown/SplunkShowDrilldown.py [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import demistomock as demisto  # noqa: F401
 from CommonServerPython import *  # noqa: F401
 
+MAX_RESULTS_TO_DISPLAY = 10  # TODO: need to remove this const (used only for demo)
```
Comment (yuvalbenshalom) [inline Packs/SplunkPy/Scripts/SplunkShowDrilldown/SplunkShowDrilldown.py:4]:
```suggestion
```

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
Looks good, see my comments.

File: Packs/SplunkPy/ReleaseNotes/3_1_28.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
+#### Integrations
+
+##### SplunkPy
+
+Added support for getting results of multiple drilldown searches enrichment associated with a Splunk notable (supported from Enterprise Security v7.2.0).
```
[the remark this answers is not in the document]
Comment (ShacharKidor) [reply Packs/SplunkPy/ReleaseNotes/3_1_28.md:6]: Added thanks!

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy_test.py [hunk 4]
```diff
@@ -1153,6 +1161,414 @@ def test_get_fields_query_part(notable_data, prefix, fields, query_part):
     assert splunk.get_fields_query_part(notable_data, prefix, fields) == query_part
 
 
+@pytest.mark.parametrize('enrichments, expected_result', [
```
[the remark this answers is not in the document]
Comment (ShacharKidor) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy_test.py:1171]: Added. 
Cool feature thanks :)

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**